### ISSUE TITLE
Why Gates::Group before Gates::Actor?

### ISSUE THREAD

**Issue by tleish** [jnunemaker/flipper#369]
Is there a reason that the sequence of gates has group gate is before the actor gate?  This seems less efficient to me.  Isn't the logic for the actor gate likely simpler and faster than the group(s) gate?

```ruby
    # Public: Get all the gates used to determine enabled/disabled for the feature.
    #
    # Returns an array of gates
    def gates
      @gates ||= [
        Gates::Boolean.new,
        Gates::Group.new,
        Gates::Actor.new,
        Gates::PercentageOfActors.new,
        Gates::PercentageOfTime.new,
      ]
    end
```

Is there a reason I'm not thinking of why you'd want to check groups before actors?

**Comment by jnunemaker** [comment]
At one point it was about network calls as I was assuming that people wouldn't be making network calls in group blocks. That probably isn't true so I could see moving actors up. Might even make sense to make groups go last. What do you think?

**Comment by tleish** [comment]
For my companies use this would make sense, however, I don't know if this would be a breaking change for existing users.

**Comment by tleish** [comment]
FYI... for now we am patching using the following:

```ruby
# frozen_string_literal: true

module Flipper
  # Change Flipper gates order for efficiency
  # TODO: Remove if Flipper gates order are updated with groups after actor
  # See: https://github.com/jnunemaker/flipper/issues/369
  module FeatureGatesPatch
    def gates
      @gates ||= begin
        group_class_index = super.map(&:class).index(::Flipper::Gates::Group)
        super.push(super.delete_at(group_class_index))
      end
    end
  end
end

Flipper::Feature.prepend Flipper::FeatureGatesPatch
```

**Comment by jnunemaker** [comment]
Closed by #370 but forgot to use the closed syntax in the PR body.
